dotnet/src/Functions/Functions.UnitTests/OpenAPI/RestApiOperationResponseTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using Microsoft.SemanticKernel.Functions.OpenAPI.Model;
+using Xunit;
+
+namespace SemanticKernel.Functions.UnitTests.OpenAPI;
+
+public class RestApiOperationResponseTests
+{
+    [Fact]
+    public void ItShouldValidateStringContentWithNoSchema()
+    {
+        //Arrange
+        var response = new RestApiOperationResponse("fake-content", "fake-content-type");
+
+        //Act
+        var result = response.IsValid();
+
+        //Assert
+        Assert.True(result);
+    }
+
+    [Fact]
+    public void ItShouldValidateByteContentTWithNoSchema()
+    {
+        //Arrange
+        var response = new RestApiOperationResponse(new byte[] { 00, 01, 02 }, "fake-content-type");
+
+        //Act
+        var result = response.IsValid();
+
+        //Assert
+        Assert.True(result);
+    }
+
+    [Fact]
+    public void ItShouldFailValidationWithSchema()
+    {
+        //Arrange
+        var response = new RestApiOperationResponse("fake-content", "fake-content-type", JsonDocument.Parse("{\"type\": \"string\"}"));
+
+        //Act
+        var result = response.IsValid();
+
+        //Assert
+        Assert.False(result); // todo -- shouldn't this validate actually?
+    }
+
+    [Fact]
+    public void ItShouldPassValidationWithSchema()
+    {
+        //Arrange
+        var response = new RestApiOperationResponse(
+            "\"fake-content\"",
+            "fake-content-type",
+            JsonDocument.Parse("{\"type\": \"string\"}"));
+
+        //Act
+        var result = response.IsValid();
+
+        //Assert
+        Assert.True(result);
+    }
+
+    [Theory]
+    [InlineData("{\"products\": [{\"id\": \"1234\", \"name\": \"Laptop\"}]}", "application/json", "{\"type\": \"object\"}")] // TODO -- Is this good enough/complete?
```

Comment:
Is the TODO at the end of this line intentional?

Reply:
Thank you! It's not. I do think I should add tests (and functionality) for the possible `ContentType` here.

Reply:
Also above on line 47 -- curious what folks think the correct behavior should be there.